src/init.c

```diff
@@ -145,14 +145,16 @@ shmem_internal_shutdown_atexit(void)
 void
 shmem_internal_start_pes(int npes)
 {
-    int tl_provided;
+    int ret, tl_provided;
 
     shmem_internal_initialized_with_start_pes = 1;
-    shmem_internal_init(SHMEM_THREAD_SINGLE, &tl_provided);
+    ret = shmem_internal_init(SHMEM_THREAD_SINGLE, &tl_provided);
+
+    if (ret) abort();
```

Comment:
FWIW, I didn't see a better way to abort here.  The runtime isn't initialized, so we can't use that to abort.  Init would have called `abort()` in this case, so that's what I did here.  One thought was adding a new `shmem_internal_abort()` that can choose how to abort based on the state of the library.